Fixed the fault tolerance when time stamp getting fails

ggtakec opened the pull request:
### Relevant Issue (if applicable)
#1872

### Details
Fixed a problem in processing when get_atime(ctime/mtime) failed to get the time.
In convert_header_to_stat, reset 0 if the above function returns -1.

gaul commented on src/cache.cpp:
```diff
@@ -786,6 +786,10 @@ bool convert_header_to_stat(const char* path, const headers_t& meta, struct stat
     }else{
+        if(mtime.tv_sec < 0){
+            mtime.tv_sec  = 0;
+            mtime.tv_nsec = 0;
```
Can you add a test to `test_read_external_object`?

ggtakec replied:
Yes, I added

ggtakec commented:
Added a test case called `test_read_external_dir_object`.

This bug occurs when `a/c/ mtime` is acquired by `stat`(or `ls`, etc) of a directory that s3fs cannot recognize.
You can check the occurrence of the problem by following the steps below:
1. Upload a new file(ex. `subdir/testfile`) containing subdirectories with `aws cli`.
2. At this point, there is no object(ex. `subdir/` for s3fs) indicating the subdirectory.
3. Run the `stat` command to the subdirectory via s3fs.
4. As a result, there was a problem that got `-1` of unixtime.

The time notation(`stat` command) for `unixtime=-1` is as follows:
```
Access: 1969-12-31 23:59:59.000000000 +0000
Modify: 1969-12-31 23:59:59.000000000 +0000
Change: 1969-12-31 23:59:59.000000000 +0000
```
With this PR fix, `a/c/mtime` for non-existent directory objects will be `unixtime=0`.

Thus, I added a test case to do this inspection.
Please be careful for this test case only checks for one `unitxtime=-1` value. (Inspects the string `1969-12-31 23:59:59.000000000`.)
